Devs-AA/Barefoot
# #168662122 Rate An Accommodation

#### What does this PR do?
Implements the functionality to allow users rate an accommodation

#### Description of Task to be completed?
- Users can rate an accommodation
- Only users who have booked the accommodation can rate it

#### How should this be manually tested?

- git clone this repo using the command `git@github.com:Devs-AA/Barefoot.git`
  - cd to the project
  - checkout to the branch using the command `git checkout ft-rate-accommodation-168662112`
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below in the request body.
- `{email: requester2@gmail.com, password: Password1$}`
2. Copy the token in the reponse
3. Make a POST request to `http:localhost:<port>/api/v1/requests` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`
 4. Send a request object containing the following for a one-way trip:
         `{
           reason: 'Business',
           departmentId: 2,
           tripType: 'oneWay',
        }`

5. Make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below  in the request body.
 `{email: barefoot2@gmail.com, password: Password1$}`
6. Make a PATCH request to `http:localhost:<port>/api/v1/requests<requestId>` with the request body containing. Note that status can only be approved or rejected
          `{
            "status": "approved"
           }`
 7. Make a POST request to `http://localhost:<port>/api/v1/accommodations/<accommodationId>`, with the same authorization header used in creating request. The request body should comply with that in the image below
![Screenshot (103)](https://user-images.githubusercontent.com/46751363/69768451-e5f8df00-1180-11ea-83e9-d8efe5f3708f.png)

8. Make a POST request to the URL `http://localhost:<PORT>/api/v1/accommodations/<accommodationId>/feedback`. with the following in the request body. Note that message is optional
![Screenshot (107)](https://user-images.githubusercontent.com/46751363/69831001-b51fb500-1227-11ea-8014-b4b78974d56d.png)

9. Try the steps above with a different user's token
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168662122](https://www.pivotaltracker.com/story/show/168662122)

#### Screenshots (if appropriate)
![Screenshot (106)](https://user-images.githubusercontent.com/46751363/69831021-d385b080-1227-11ea-8477-004f54d688ed.png)
![Screenshot (104)](https://user-images.githubusercontent.com/46751363/69831026-dd0f1880-1227-11ea-8a5f-623cc39a4d35.png)
![Screenshot (105)](https://user-images.githubusercontent.com/46751363/69831028-e0a29f80-1227-11ea-9a5e-5bd1b4319117.png)

#### Questions
N/A
